src/prometheus_server.py

```diff
@@ -33,7 +59,7 @@ def reload_configuration(self) -> bool:
         Returns:
           True if reload succeeded (returned 200 OK); False otherwise.
         """
-        url = f"{self.base_url}/-/reload"
+        url = urljoin(self.base_url, "-/reload")
```

Comment:
Nitpick but worth thinking. Typically I would expect a programmer will write a URL path starting with a slash eg. `/home_dir` rather than `home`. While I see no harm in adding a terminating slash to `self.base_url`, I do not see why remove it from `/-/reload`. `urljoin` is perfectly fine with having both terminating slash in `self.base_url` and `/-/reload`.

Reply:
> urljoin is perfectly fine with having both terminating slash in self.base_url and /-/reload

I had tried it. `urljoin` seems to disagree with your assessment :D

Reply:
```
Python 3.9.5 (default, Nov 18 2021, 16:00:48) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import urljoin
>>> urljoin("https://ubuntu.com/michele/is/right","/balbir/is/wrong")
'https://ubuntu.com/balbir/is/wrong'
```

Yes I am being petty here :D

Reply:
```
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import urljoin
>>> urljoin("https://ubuntu.com/michele/is/right","/balbir/is/wrong")
'https://ubuntu.com/balbir/is/wrong'
>>> urljoin("https://ubuntu.com/michele/is/right/","/balbir/is/wrong")
'https://ubuntu.com/balbir/is/wrong'
>>> 
```
This is what I am talking about.

Reply:
I see what you mean.